src/main/java/somethingrandom/entity/Item.java

```diff
@@ -1,4 +1,60 @@
 package somethingrandom.entity;
 
+import java.time.Instant;
+import java.util.UUID;
+
+/**
+ * An abstract Item.
+ */
 public abstract class Item {
+    /**
+     * The Name.
+     */
+    public String name;
+    /**
+     * The ID.
+     */
+    public UUID id;
+    /**
+     * The Creation date.
+     */
+    public Instant creationDate;
+    /**
+     * The Item kind.
+     */
+    public ItemKind itemKind;
+
+    /**
+     * Gets creation date.
+     *
+     * @return the creation date
+     */
+    public abstract Instant getCreationDate();
```

Comment:
I think this should be concrete instead of abstract, just implement it as `{ return creationDate; }` here.